### PR TITLE
[Peer Review] Tests for ConfigureVoltageLevel and ConfigureCurrentLevel APIs

### PR DESCRIPTION
**Pull request by ni-pkhare**
### What does this Pull Request accomplish?

Tests for 2 APIs:
1) ConfigureVoltageLevel
2) ConfigureCurrentLevel

### Why should this Pull Request be merged?

Both of these APIs first require configuring the Output Function in the specified channel.
So tests for both the APIs also include configuring the output function in the beginning and assertion for output function value check in the end.

### What testing has been done?

Manually ran the tests.

### CODE REVIEW

File: source/tests/system/nidcpower_driver_api_tests.cpp [hunk 1]
```diff
@@ -140,6 +187,48 @@ TEST_F(NiDCPowerDriverApiTest, NiDCPowerReset_SendRequest_ResetCompletesSuccessf
   expect_api_success(response.status());
 }
 
+TEST_F(NiDCPowerDriverApiTest, ConfigureOutputFunctionAndVoltageLevel_ConfiguresSuccessfuly)
+{
+  const char* channel_name = "0";
+  ViReal64 expected_voltage_level = 3.0;
+  configure_output_function(channel_name, dcpower::OutputFunction::OUTPUT_FUNCTION_NIDCPOWER_VAL_DC_VOLTAGE);
```
Comment (kruthi-NI) [inline source/tests/system/nidcpower_driver_api_tests.cpp]: The dcpower::OutputFunction::OUTPUT_FUNCTION_NIDCPOWER_VAL_DC_VOLTAGE can be extracted to a local variable something like expected_output_function and used in the assert statements below also

File: source/tests/system/nidcpower_driver_api_tests.cpp [hunk 2]
```diff
@@ -140,6 +187,48 @@ TEST_F(NiDCPowerDriverApiTest, NiDCPowerReset_SendRequest_ResetCompletesSuccessf
...
+  EXPECT_TRUE(status.ok());
+  expect_api_success(response.status());
+
+  ViReal64 actual_voltage_level = get_real64_attribute(channel_name, dcpower::NiDCPowerAttributes::NIDCPOWER_ATTRIBUTE_VOLTAGE_LEVEL);
+  ViInt32 output_function_value = get_int32_attribute(channel_name, dcpower::NiDCPowerAttributes::NIDCPOWER_ATTRIBUTE_OUTPUT_FUNCTION);
+  EXPECT_EQ(expected_voltage_level, actual_voltage_level);
+  EXPECT_EQ(dcpower::OutputFunction::OUTPUT_FUNCTION_NIDCPOWER_VAL_DC_VOLTAGE, output_function_value);
+}
+
+TEST_F(NiDCPowerDriverApiTest, ConfiguresOutputFunctionAndCurrentLevel_ConfiguresSuccessfuly)
```
Comment (kruthi-NI) [inline source/tests/system/nidcpower_driver_api_tests.cpp]: There is a typo. Test name should have been ConfigureOutputFunctionAndCurrentLevel_ConfiguresSuccessfully

File: source/tests/system/nidcpower_driver_api_tests.cpp [hunk 3]
```diff
@@ -140,6 +187,48 @@ TEST_F(NiDCPowerDriverApiTest, NiDCPowerReset_SendRequest_ResetCompletesSuccessf
...
+{
+  const char* channel_name = "0";
+  ViReal64 expected_current_level = 3.0;
+  configure_output_function(channel_name, dcpower::OutputFunction::OUTPUT_FUNCTION_NIDCPOWER_VAL_DC_CURRENT);
```
Comment (kruthi-NI) [inline source/tests/system/nidcpower_driver_api_tests.cpp]: dcpower::OutputFunction::OUTPUT_FUNCTION_NIDCPOWER_VAL_DC_CURRENT can be extracted to a local variable and used below

**Review by kruthi-NI — APPROVED** [review]

File: source/tests/system/nidcpower_driver_api_tests.cpp (hunk 3 above)
Comment (ni-pkhare) [reply source/tests/system/nidcpower_driver_api_tests.cpp]: Done.

File: source/tests/system/nidcpower_driver_api_tests.cpp (hunk 2 above)
Comment (ni-pkhare) [reply source/tests/system/nidcpower_driver_api_tests.cpp]: Done.

File: source/tests/system/nidcpower_driver_api_tests.cpp (hunk 1 above)
Comment (ni-pkhare) [reply source/tests/system/nidcpower_driver_api_tests.cpp]: Done.

File: CMakeLists.txt [hunk 4]
```diff
@@ -300,6 +300,7 @@ add_executable(SystemTestsRunner
     "source/tests/system/niswitch_driver_api_tests.cpp"
     "source/tests/system/niswitch_session_tests.cpp"
     "source/tests/system/nisync_session_tests.cpp"
+    "source/tests/system/nidcpower_driver_api_tests.cpp"
```
Comment (ganeshnie) [inline CMakeLists.txt]: I think this file is already part of the features/ni-dcpower branch. I think the diff in this PR is not updated?

Comment (ni-pkhare) [reply CMakeLists.txt:303]: Correct. I planned to change the target of this PR to feature branch. Its just that I was waiting for the current target branch to get submitted to the feature branch.

Comment (ni-pkhare) [reply CMakeLists.txt]: @ganeshnie There were some merge conflicts that I could not resolve so I would close this PR and create a new PR which does the same but is directly branched out from feature branch.
New PR : [#23](https://github.com/doshirohan/grpc-device/pull/23)

**Pull request closed without merging**